nservicebus/pipeline/steps-stages-connectors_outgoing_core_[6.0,).partial.md

```diff
@@ -3,13 +3,14 @@
 * Operation specific processing: There is a dedicated stage for each context operation (e.g. Send, Publish, Subscribe, ...). Behaviors can use one of the following contexts: `IOutgoingSendContext`, `IOutgoingPublishContext`, `IOutgoingReplyContext`, `ISubscribeContext`, `IUnsubscribeContext`. Subscribe and Unsubscribe are not shown on the diagram below.
 * Outgoing Logical Message: Behaviors on this stage have access to the message which should be sent. Use `IOutgoingLogicalMessageContext` in a behavior to enlist in this stage.
 * Outgoing Physical Message: Enables to access the serialized message. This stage provides `IOutgoingPhysicalMessageContext` to it's behaviors.
-* Routing: Provides access to the routing strategies that have been selected for the outgoing message. This stage provides `IRoutingContext` to it's behaviors.
-* Batch Dispatch: When messages are sent as part of a message handler or saga handler, or when using `IMessageSession` to send messages. Outgoing messages are [collected into a batch](/nservicebus/messaging/batched-dispatch.md) and handed to the Batch Dispatch stage all at once once message processing has been completed. This stage provides access to the collection of transport operations that are to be dispatched. This stage provides `IBatchDispatchContext` to it's behaviors. The batch dispatch stage can be bypassed by specifying [immediate dispatch](/nservicebus/messaging/send-a-message.md) for an outgoing message.
+* Routing: Allows the selected routing strategies for outgoing messages to be manipulated. If the outgoing pipeline was initiated by the incoming pipeline, this stage collects [all outgoing operations](/nservicebus/messaging/batched-dispatch.md) (except for [immediate dispatch messages](/nservicebus/messaging/send-a-message.md#dispatching-a-message-immediately)).
```

Comment:
I think this is not true. The routing stage is executed "synchronously" and the result of it is collected in https://github.com/Particular/NServiceBus/blob/master/src/NServiceBus.Core/Pipeline/Outgoing/RoutingToDispatchConnector.cs and passed to the batched dispatch.

Reply:
We are only calling the dispatch stage if the operations wasn't batched? https://github.com/Particular/NServiceBus/blob/master/src/NServiceBus.Core/Pipeline/Outgoing/RoutingToDispatchConnector.cs#L37